From 71a87f5c870d201bfef08aed6b9d56464d5fc06f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alex=20R=C3=B8nne=20Petersen?= <alex@alexrp.com>
Date: Thu, 4 Sep 2025 01:13:27 +0200
Subject: [PATCH 090/133] std.elf: add new EM values from gABI 4.3
X-Developer-Signature: v=1; a=openpgp-sha256; l=1203; i=xtex@astrafall.org;
 h=from:subject; bh=HmpfstWCQIaVeryfTcfv4lOZY8TfQ32x/SkxDfXTX1s=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78jj6v320UhI9vUH+d9al5jUfpYpzeVcfajfSEf6wY
 7r4mxLOjoksDGJcDJZiiixFhg3erDrp/KLLymVh5rAygQyRFmlgYGBgYGHgy03MKzXSMdIz1TbU
 MzTUMdIxYuDiFICpPrWL4X/xM/Esx6bHp89LLBD6sPt/7Py+Vofny3w8ZbJ0arecOjqD4Q8/517
 PDnVbpeNSB6rE7SQy21/dzw649/W2QY/66bqULlYA
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

(cherry picked from commit 9e076d8c7548eacc0bbbad770e5e06c3a427c687)
---
 lib/std/elf.zig | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/lib/std/elf.zig b/lib/std/elf.zig
index 24545171eeaa..b0e5dd8fec5c 100644
--- a/lib/std/elf.zig
+++ b/lib/std/elf.zig
@@ -1377,6 +1377,8 @@ pub const EM = enum(u16) {
     TI_C2000 = 141,
     /// Texas Instruments TMS320C55x DSP family
     TI_C5500 = 142,
+    /// Texas Instruments Application Specific RISC Processor, 32bit fetch
+    TI_ARP32 = 143,
     /// Texas Instruments Programmable Realtime Unit
     TI_PRU = 144,
     /// STMicroelectronics 64bit VLIW Data Signal Processor
@@ -1537,6 +1539,18 @@ pub const EM = enum(u16) {
     TACHYUM = 261,
     /// NXP 56800EF Digital Signal Controller (DSC)
     @"56800EF" = 262,
+    /// Solana Bytecode Format
+    SBF = 263,
+    /// AMD/Xilinx AIEngine architecture
+    AIENGINE = 264,
+    /// SiMa MLA
+    SIMA_MLA = 265,
+    /// Cambricon BANG
+    BANG = 266,
+    /// Loongson LoongGPU
+    LOONGGPU = 267,
+    /// Wuxi Institute of Advanced Technology SW64
+    SW64 = 268,
     /// AVR
     AVR_OLD = 0x1057,
     /// MSP430
-- 
2.52.0

